src/qml/nodemodel.cpp

```diff
@@ -27,14 +28,60 @@ void NodeModel::setBlockTipHeight(int new_height)
     }
 }
 
+void NodeModel::setRemainingSyncTime(double new_progress)
+{
+    int currentTime = QDateTime::currentDateTime().toMSecsSinceEpoch();
+
+    // keep a vector of samples of verification progress at height
+    m_block_process_time.push_front(qMakePair(currentTime, new_progress));
+
+    // show progress speed if we have more than one sample
+    if (m_block_process_time.size() >= 2) {
+        double progressDelta = 0;
+        int timeDelta = 0;
+        int remainingMSecs = 0;
+        double remainingProgress = 1.0 - new_progress;
+        for (int i = 1; i < m_block_process_time.size(); i++) {
+            QPair<int, double> sample = m_block_process_time[i];
+
+            // take first sample after 500 seconds or last available one
```

Comment:
but `currentTime` was set in milliseconds in line 33, now we're doing math with seconds values?